### ISSUE TITLE
s-format: Wrong type argument: arrayp, nil

### ISSUE THREAD

**Issue by mpomeran** [tmalsburg/helm-bibtex#203]
Hi, 
Thanks for the wonderful program. Alas, I'm having some trouble with it. 
I upgraded to 25.3 and some of my packages and now none of my bibliographies 
seem to be loading properly. 
I get the following error message each time I load helm-bibtex: 

Parsing bibliography file ~/Dropbox/BIBLIO/main.bib ...
Resolving cross-references ...
s-format: Wrong type argument: arrayp, nil

I've winnowed down my customizations to (require 'helm-bibtex)

(setq bibtex-completion-bibliography '("~/Dropbox/BIBLIO/main.bib"))

I checked my .bib file and doesn't seem to be the culprit... 

Hmmm... 

Thanks

**Comment by mpomeran** [comment]
I ran (length (helm-completion-candidates) and all of the bibtex entries are loaded...

**Comment by mpomeran** [comment]
fixed it!! It was an error in my customization file.

**Comment by tmalsburg** [comment]
Great, thanks for the update.

**Comment by meliache** [comment]
I had exactly this error when using [org-roam-bibtex](https://github.com/org-roam/org-roam-bibtex) (in particular the `orb-note-actions` command), which uses `bibtex-completion-format-entry` internally. The cause was that org-roam-bibtex never called `(bibtex-completion-init)`, therefore `bibtex-completion-display-formats-internal` was never set and thus bibtex tried using `nil` as a format-string, which obviously failed.

When I first got the error, I ran the debugger to find that the error occurs in `bibtex-completion`. I had installed `bibtex-completion` explicitly with `straight.el`, but neither of `helm-bibtex` or `ivy-bibtex`, since I just use `citar`. Through some trial and error I found that I don't get the error when running `ivy-bibtex` first, so I looked into its code and found it calls `(bibtex-completion-init)`.

Adding that initialization function call to my init-file fixed the issue. However, it's not command that's documented anywhere in the installation or configuration instructions, so I assume it was not a user error of mine but a bug in `org-roam-bibtex`, so I also opened a PR to fix that: https://github.com/org-roam/org-roam-bibtex/pull/256

Anyway, I also decided to comment here because I was a bit frustrated when I read
>  It was an error in my customization file.
but with no mention of what the error was, so that information was not useful to me. I hope, this comment might be useful for someone who might also stumble upon this error in the future.
